File: app/src/main/java/ru/sukharev/myrecipes/model/IModel.java
package ru.sukharev.myrecipes.model;

import java.util.List;

import ru.sukharev.myrecipes.pojo.Recipe;

/**
 * Interface describing user interaction with model layer
 */

public interface IModel {

    void getRecipes(GetRecipesCallback callback);

    void addRecipe(Recipe recipe, AddRecipeCallback callback);

    interface GetRecipesCallback {

        void onResult(List<Recipe> recipes);
    }

    interface AddRecipeCallback{

        void onSuccess(long result);

        void onFail();
    }
}

File: app/src/main/java/ru/sukharev/myrecipes/model/IModel.kt
package ru.sukharev.myrecipes.model

import ru.sukharev.myrecipes.pojo.Recipe

/**
 * Interface describing user interaction with model layer
 */

interface IModel {

    fun getRecipes(callback: GetRecipesCallback)

    fun addRecipe(recipe: Recipe, callback: AddRecipeCallback)

    interface GetRecipesCallback {

        fun onResult(recipes: List<Recipe>)
    }

    interface AddRecipeCallback {

        fun onSuccess(result: Long)

        fun onFail()
    }
}
